src/main/scala/org/broadinstitute/dsde/workbench/sam/util/OpenCensusIOUtils.scala

```diff
@@ -1,34 +1,53 @@
 package org.broadinstitute.dsde.workbench.sam.util
 
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.Directives._
 import io.opencensus.trace.{Span, Status}
 import io.opencensus.scala.Tracing._
-
+import io.opencensus.scala.akka.http.TracingDirective.traceRequest
 import cats.effect.IO
 
 object OpenCensusIOUtils {
 
-  def traceIOWithParent[T](
-                            name: String,
-                            parentSpan: Span,
-                            failureStatus: Throwable => Status = (_: Throwable) => Status.UNKNOWN
-                          )(f: Span => IO[T]): IO[T] =
-    traceIOSpan(IO(startSpanWithParent(name, parentSpan)), failureStatus)(f)
+  def traceIOWithContext[T](
+                             name: String,
+                             samRequestContext: SamRequestContext,
+                             failureStatus: Throwable => Status = (_: Throwable) => Status.UNKNOWN
+                          )(f: SamRequestContext => IO[T]): IO[T] = {
+    if (samRequestContext.parentSpan == null) { // ignore calls from unit tests and calls on boot; an alternative here would be to have a MockOpenCensusIOUtils to unit tests
```

Comment:
I'd be interested to know if anyone thinks we should use something like MockOpenCensusIOUtils for unit tests instead of checking for null

Reply:
why is it null instead of None?
I don't think we care about asserting anything about the traces, but if it's painless to use a fake trace, that's better than None since the fake trace is "closer" to reality for testing adding spans. But if it's much work at all, not worth it imo.